src/utils/contractCall/mutableFunctionCall.ts

```diff
@@ -0,0 +1,54 @@
+import { ethers } from "ethers";
+import { ExtensionContext } from "vscode";
+import { logger } from "../../lib";
+import { EstimateGas, useContractType } from "../../types";
+import { getGasEstimates } from "../functions";
+import {
+  getConfiguration,
+  getSelectedNetConf,
+  getSignedContract,
+} from "../networks";
+
+export const MutableFunctionCall = async (
+  context: ExtensionContext,
+  state: string,
+  params: any[],
+  useContract: useContractType
+) => {
+  const MAX_FEE_PER_GAS = 100;
+  const { abiItemName, contractAddress, contractName } = useContract;
+
+  const contract = await getSignedContract(context, contractAddress);
+
+  let gasCondition = (await context.workspaceState.get("gas")) as string;
+
+  const gasEstimate: EstimateGas | undefined = await getGasEstimates(
+    gasCondition,
+    context
+  );
+
+  const maxFeePerGas =
+    gasEstimate !== undefined ? gasEstimate.maxFeePerGas : MAX_FEE_PER_GAS;
+
+  const settingsGasLimit = (await getConfiguration().get("gasLimit")) as number;
+
+  const value =
```

Comment:
When we create the input `JSON` for payable functions create the `JSON` with a value field. We should get all the necessary parameters from the input `JSON. Also optionally keep this input.